packages/hebao_v1/contracts/modules/core/ForwarderModule.sol

```diff
@@ -137,17 +161,12 @@ abstract contract ForwarderModule is BaseModule
             metaTx.gasToken,
             metaTx.gasPrice,
             metaTx.gasLimit,
-            metaTx.data,
+            data,
             signature
         );
 
-        // Nonce update must come after the real transaction in case of new wallet creation.
-        if (metaTx.nonce != 0) {
```

Comment:
The original comment points out the verify must come after the real transaction. Is it safe to move it ahead?

Reply:
No interaction necessary with the store any more which needed a valid wallet, so it shouldn't matter any more where we update the nonce for that.